lib/compilers/flang.ts

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2022, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import {FortranCompiler} from './fortran';
+
+export class FlangCompiler extends FortranCompiler {
+    static override get key() {
+        return 'flang';
+    }
+
+    override getDefaultExecOptions() {
+        const result = super.getDefaultExecOptions();
+        const gfortranPath = this.compilerProps('gfortranPath');
+        if (gfortranPath) {
+            result.env.PATH = result.env.PATH + ':' + gfortranPath;
```

Comment:
Shouldn't we try to use `join()` from `os.path` method instead of string concat with `:` ?

Reply:
`path.join()` is to make 1 absolute paths from a root and a relative path. That would not work here.

Reply:
https://nodejs.org/docs/latest/api/path.html#pathdelimiter  :D

Reply:
Thanks for the tip, will implement as you suggest :)

Reply:
We have `:` and `;` in a few places, maybe we could use this somewhere else too!